### PR TITLE
Refined Python support plus more args

### PR DESCRIPTION
**Pull request by emeryberger**

### CODE REVIEW

File: src/cwhy/cwhy.py [hunk 1]
```diff
@@ -118,24 +118,27 @@ def evaluate(args, stdin):
 
 def evaluate_prompt(args, prompt, wrap=True):
     if args["show_prompt"]:
+        import sys
```
Comment (nicovank) [inline src/cwhy/cwhy.py:121]:
```suggestion
```

Already imported.

File: src/cwhy/cwhy.py [hunk 2]
```diff
@@ -118,24 +118,27 @@ def evaluate(args, stdin):
...
+        
         diagnostic_lines = diagnostic.splitlines()
 
         self.code_locations = {}
 
-        # Magic number - don't send more than this many code locations.
+        # Don't send more than this many code locations.
         # This is just to prevent overwhelming OpenAI.
-        max_code_locations = 10
+        max_code_locations = int(args["max_context"])
```
Comment (nicovank) [inline src/cwhy/cwhy.py:141]:
```suggestion
        max_code_locations = args["max_context"]
```

Nit, this is already an int from argparse's `type=int`.

File: src/cwhy/cwhy.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def read_lines(file_path: str, start_line: int, end_line: int) -> (str, int):
         # return the requested lines as a list
         return ("\n".join(lines[start_line:end_line]) + "\n", start_line, end_line)
     except FileNotFoundError:
-        print(f"Cwhy warning: file not found: {file_path.lstrip()}")
+        # print(f"Cwhy warning: file not found: {file_path.lstrip()}")
```
Comment (nicovank) [inline src/cwhy/cwhy.py:81]: Any reason to remove this warning? I've found it helpful for C/C++. Does it come up a lot in Java/Python?

Comment (emeryberger) [reply src/cwhy/cwhy.py:81]: I may have eliminated it by changing the regex for Python (which was including internals like `<string>`). OK to restore it. (Why is it useful, though?)

Comment (nicovank) [reply src/cwhy/cwhy.py:81]: When running in non-wrapper mode, non-found files are relatively common. I guess it's just good to know that the source of that file file will not be included in the prompt.

Also for reference, I think we are not properly sending nothing in the prompt when a file is not found, and we may include some garbage, will look into that.

**Review by nicovank — APPROVED** [review]

File: src/cwhy/cwhy.py (hunk 3 above)
Comment (emeryberger) [reply src/cwhy/cwhy.py:81]: I also suspect there is an off-by-one error in the reported line numbers for code snippets.

Comment (nicovank) [reply src/cwhy/cwhy.py:81]: FYI, both issues now fixed in 61e374f4c8563ccdd6a300cc4c21725ab7fe9987.

**Pull request merged**